app/crud/versions.py

```diff
@@ -31,14 +35,22 @@ async def get_version_names(dataset: str) -> List[Any]:
 
 
 async def get_version(dataset: str, version: str) -> ORMVersion:
-    row: ORMVersion = await ORMVersion.get([dataset, version])
+    row: ORMVersion = (
+        await ORMVersion.load(metadata=ORMVersionMetadata)
+        .where(ORMVersion.dataset == dataset)
+        .where(ORMVersion.version == version)
+        .gino.first()
+    )
+
     if row is None:
         raise RecordNotFoundError(
             f"Version with name {dataset}.{version} does not exist"
         )
-    d: ORMDataset = await datasets.get_dataset(dataset)
 
-    return update_metadata(row, d)
+    if getattr(row, "metadata", None) is None:
+        row.metadata = {}
```

Comment:
This is fine, but would it be just as well (and slightly more succinct) to do
row.metadata = getattr(row, "metadata", {})
? Or could metadata be None anyway?

Reply:
Duh, updated.  The pydantic model not liking `None` so updated with `{}` default.